File: Python/percentage_change.py
import time
import logging
import datetime

logging.basicConfig(filename = "/home/champ/problems2.log", level = logging.DEBUG)
logger = logging.getLogger()


def pchg():
    cv = int(input("Please provide the current value: "))
    iv = int(input("Please provide the initial value: "))

    c_minus_i = cv - iv
    try:
        c_minus_i = c_minus_i / iv
    except ZeroDivisionError:
        logger.error(f"{datetime.datetime.now()} - You cannot divide by zero")
        raise
    finally:
        logger.info(f"This app finished at {datetime.datetime.now()}")

    return c_minus_i * 100


if __name__ == "__main__":
    pc = pchg()
    print("The percentage change is {}".format(pc))

File: Go/percentage_change.go
package main

import (
	"fmt"
	"time"
)


func main(){
	
	var currentValue float32
	var initialValue float32

	fmt.Printf("Please enter the current value: ")
	_, err := fmt.Scanf("%f", &currentValue)

	if err != nil {
		fmt.Println(err)
	}

	fmt.Printf("Please enter the initial value: ")
	_, err2 := fmt.Scanf("%f", &initialValue)

	if err2 != nil {
		fmt.Println(err2)
	}

	fmt.Printf("The percent change is: %.2f%%\n", percent_change(currentValue, initialValue))

	fmt.Println(time.Now())
}

func percent_change( cv float32, iv float32) float32 {
	return (cv - iv) / iv * 100

}

